presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -432,6 +425,8 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
 
         long transactionId = shardManager.beginTransaction();
 
+        setTransactionId(transactionId);
```

Comment:
This looks like it should work, but it seems to fit the model better by having the shardManager.beginTransaction() to be moved to the connector beginTransaction() call and store the transactionId in the transaction handle?